### PR TITLE
[radio] Add LNA gain offset to the RSSI value

### PR DESCRIPTION
**Pull request by Irving-cl**
The offset of LNA gain should be removed when reporting RSSI.

### CODE REVIEW

File: src/src/radio.c [hunk 1]
```diff
@@ -1006,10 +1008,15 @@ void nrf_802154_received_timestamp_raw(uint8_t *p_data, int8_t power, uint8_t lq
 
     receivedFrame->mPsdu               = &p_data[1];
     receivedFrame->mLength             = p_data[0];
-    receivedFrame->mInfo.mRxInfo.mRssi = power;
+    receivedFrame->mInfo.mRxInfo.mRssi = power - sLnaGain;
     receivedFrame->mInfo.mRxInfo.mLqi  = lqi;
     receivedFrame->mChannel            = nrf_802154_channel_get();
 
+    if (receivedFrame->mInfo.mRxInfo.mRssi > 0)
```
Comment (zhanglongxia) [inline src/src/radio.c:1015]: I think we need to limit both the upper and lower bounds. Thoughts?

**Review by zhanglongxia — APPROVED** [review]
LGTM

**Review by lmaciejonczyk — COMMENTED** [review]
What is the reason of this change since this is contrary to what has been agreed here: https://github.com/openthread/openthread/pull/5563#issuecomment-701740884 ?
What more if you are going to add LNA gain offset to RSSI then you should increase receive sensitivity for the purpose of link margin correct calculation.
@Irving-cl Thoughts?

**Comment by Irving-cl** [comment]
Need more discussion.

**Pull request closed without merging**